hooks/checkout

```diff
@@ -140,6 +174,18 @@ checkout() {
   git reset --hard
   git clean -ffxdq
 
+  local proxy_url
+  proxy_url=$(print_proxy_url)
+  if [[ -n "${proxy_url:-}" ]]; then
+    log_info "Using proxy: ${proxy_url}"
+    git config http.proxy "${proxy_url}"
+    git config url."https://github.com/".insteadOf git@github.com:
```

Comment:
We might need to explicitly set `protocol.version` to 2, it didn't become a default until quite recent versions of git.

Reply:
done